src/AudioToolbox/MusicTrack.cs

```diff
@@ -129,6 +129,20 @@ internal override IntPtr ToUnmanaged ()
 
 	public class MusicEventUserData : MidiRawData {
 		public MusicEventUserData () {}
+
+		internal MusicEventUserData (IntPtr handle)
+		{
+			var buffer = new byte[sizeof(int)];
+			Marshal.Copy (handle, buffer, 0, sizeof(int));
+			int length = BitConverter.ToInt32 (buffer, 0);
```

Comment:
Can't this be replaced with this code?

    int length = Marshal.ReadInt32 (handle);